codebundles/k8s-deployment-healthcheck/deployment_logs.sh

```diff
@@ -0,0 +1,197 @@
+# fields and the format - need to figure out how to best match the right formats, 
+# or can we just use logline
+
+##### Begin query #####
+
+# Format file / table http_logrus_custom
+# Search for http log format used by online-boutique (which uses logrus but is custom)
+for FILE in "${LOG_FILES[@]}"; do
+    echo "$FILE"
+    LOG_SUMMARY=$(lnav -n -c ';SELECT COUNT(*) AS error_count, CASE WHEN "http.req.path" LIKE "/product%" THEN "/product" ELSE "http.req.path" END AS root_path, "http.resp.status" FROM http_logrus_custom WHERE "http.resp.status" = 500 AND NOT "http.req.path" = "/" GROUP BY root_path, "http.resp.status" ORDER BY error_count DESC;' $FILE)
+    echo "$LOG_SUMMARY"
+    INTERESTING_PATHS+=$(echo "$LOG_SUMMARY" | awk 'NR>1 && NR<5 {sub(/^\//, "", $2); print $2}')$'\n'
+done
+
+if [[ -n "$INTERESTING_PATHS" ]]; then
+    SEARCH_RESOURCES=$(echo "$INTERESTING_PATHS" | awk -F'/' '{for (i=1; i<=NF; i++) print $i}' | sort | uniq)
+    issue_descriptions+=("HTTP Errors found for paths: $SEARCH_RESOURCES")
+else
+    echo "No interesting paths found."
+fi
+##### End query #####
+
+
+
+
+# # Fetch a list of all resources in the namespace
+## Heavyweight - this times out after 30s, but is a better way to get any and all resources
+# SEARCH_LIST=$(${KUBERNETES_DISTRIBUTION_BINARY} api-resources --verbs=list --namespaced -o name  | xargs -n 1 ${KUBERNETES_DISTRIBUTION_BINARY} get --show-kind --ignore-not-found -n $NAMESPACE)
+
+## Lightweight - we explicitly specify which resources we want to search
+# Run RESOURCE_SEARCH_LIST only if SEARCH_RESOURCES has content
+if [[ -n "$SEARCH_RESOURCES" ]]; then
+    RESOURCE_SEARCH_LIST=$(${KUBERNETES_DISTRIBUTION_BINARY} get deployment,pods,service,statefulset --context=${CONTEXT} -n ${NAMESPACE})
+else
+    echo "No search queries returned results."
+    exit
+fi
+
+
+EVENT_SEARCH_LIST=$(${KUBERNETES_DISTRIBUTION_BINARY}  get events --context=${CONTEXT} -n ${NAMESPACE})
+event_details="\nThe namespace ${NAMESPACE} has produced the following interesting events:"
+event_details+="\n"
+
+# For each value, search the namespace for applicable resources
+INTERESTING_RESOURCES=""
+for RESOURCE in "${SEARCH_RESOURCES[@]}"; do 
+    event_details+=$(echo "$EVENT_SEARCH_LIST" | grep "$RESOURCE" | grep -Eiv "Normal")
+    INTERESTING_RESOURCES+=$(echo "$RESOURCE_SEARCH_LIST" | grep "$RESOURCE")
+done
+
+# Try to generate some recommendations from the resource strings we discovered
+recommendations=()
+if [[ -n "$INTERESTING_RESOURCES" ]]; then
+    while read -r line; do
+        # Splitting columns into array
+        IFS=' ' read -ra cols <<< "$line"
+        resource="${cols[0]}"
+        status="${cols[1]}"
+        restarts="${cols[3]}"
+
+        # Extracting resource type and name
+        IFS='/' read -ra details <<< "$resource"
+        type="${details[0]}"
+        name="${details[1]}"
+
+        case "$type" in
+        pod)
+            if [[ "$status" != "Running" ]]; then
```

Comment:
nitpick: the format the platform is expecting for nextsteps submissions is:
```
task titles, separate by commas,
\n
\n
{kind}:{name}
{kind}:{name}
```

Reply:
I'm specifically not using next steps here; I prefer the output from the recommendations at this point in time.

Reply:
I'm not sure that I need to follow that format really... I've tested the submitted recommendations and they adequately provide new search results. The format above, to me, is not as human readable. The recommendations I've provided are very specific, 1 per line item, recommendations - whereas the format above looks like it's all tasks first, and then service and objects. As a reader of the output, I want to know which tasks apply to which objects.